src/wallet/mod.rs

```diff
@@ -836,23 +838,40 @@ where
             current_height,
         )?;
 
-        // get drain script
-        let drain_script = match params.drain_to {
-            Some(ref drain_recipient) => drain_recipient.clone(),
-            None => self
-                .get_internal_address(AddressIndex::New)?
-                .address
-                .script_pubkey(),
+        // prepare the drain script
+        let weighted_drain_script = {
+            let script_pubkey = match params.drain_to {
+                Some(ref drain_recipient) => drain_recipient.clone(),
+                None => self
+                    .get_internal_address(AddressIndex::New)?
+                    .address
+                    .script_pubkey(),
+            };
+
+            let satisfaction_weight = if params.drain_to.is_none() {
+                self.get_descriptor_for_keychain(KeychainKind::Internal)
+                    .max_satisfaction_weight()
+                    .unwrap()
+            } else {
+                0
+            };
+
+            WeightedScript {
+                satisfaction_weight,
+                script_pubkey,
+            }
         };
 
-        let coin_selection = coin_selection.coin_select(
+        let coin_selection = get_selection(
+            coin_selection,
             self.database.borrow().deref(),
             required_utxos,
             optional_utxos,
             fee_rate,
             outgoing + fee_amount,
-            &drain_script,
+            &weighted_drain_script,
         )?;
+
         fee_amount += coin_selection.fee_amount;
         let excess = &coin_selection.excess;
```

Comment:
Nit: We can get rid of the `excess` variable.

Reply:
Do you mean to replace all occurrences of `excess` with `coin_selection.excess`?